### PR TITLE
Handle distribution by language

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/47f6e2704731852d6a5e8960222e161774a87f73/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "index" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/47f6e2704731852d6a5e8960222e161774a87f73/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "not_analyzed" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ProjectMeasuresIndex.java#L200](https://github.com/SonarSource/sonarqube/blob/47f6e2704731852d6a5e8960222e161774a87f73/server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java#L200): Define a constant instead of duplicating this literal "nested_" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/21483c73950aa38191a469443a4c9088745f9149/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "index" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/21483c73950aa38191a469443a4c9088745f9149/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "not_analyzed" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ProjectMeasuresIndex.java#L200](https://github.com/SonarSource/sonarqube/blob/21483c73950aa38191a469443a4c9088745f9149/server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java#L200): Define a constant instead of duplicating this literal "nested_" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/FilterParser.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.StreamSupport;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.core.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class FilterParser {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/FilterParser.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1118)

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java [hunk 2]
```diff
@@ -207,8 +217,10 @@ private SearchResults searchData(DbSession dbSession, SearchProjectsRequest requ
   }
 
   @CheckForNull
-  private Set<String> buildFilterOnFavoriteProjectUuids(List<String> criteria, Set<String> favoriteProjectUuids) {
-    if (hasIsFavoriteCriterion(criteria)) {
+  private Set<String> buildFilterOnFavoriteProjectUuids(List<FilterParser.Criterion> criteria, Set<String> favoriteProjectUuids) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java]: MAJOR Make "buildFilterOnFavoriteProjectUuids" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/7efa889c74c372251339a09e9aeee7782519cd74/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "index" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [NewIndex.java#L277](https://github.com/SonarSource/sonarqube/blob/7efa889c74c372251339a09e9aeee7782519cd74/server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java#L277): Define a constant instead of duplicating this literal "not_analyzed" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ProjectMeasuresIndex.java#L200](https://github.com/SonarSource/sonarqube/blob/7efa889c74c372251339a09e9aeee7782519cd74/server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java#L200): Define a constant instead of duplicating this literal "nested_" 3 times. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by julienlancelot** [comment]
I'm closing this PR as it has been merged into https://github.com/SonarSource/sonarqube/pull/1691

**Pull request closed without merging**